package/MDAnalysis/core/groups.py

```diff
@@ -338,8 +343,25 @@ def __init__(self, *args):
         self._u = u
         self._cache = dict()
 
+    def __setattr__(self, attr, value):
+        # `ag.this = 42` calls setattr(ag, 'this', 42)
+        # we scan 'this' to see if it is either 'private'
+        # or a known attribute (WHITELIST)
+        if (not attr.startswith('_') and
+            # Don't change this to isinstance,
+            # it is literally meant, if this class is the same
+            # Subclasses are free to setattr as they wish
+            type(self) in (self.universe._classes[AtomGroup],
```

Comment:
Yes. I was more thinking the mechanism itself is non standard so it would avoid surprises my removing it. It's not a real feature that provides functionality, more of a safety mechanism